From 74ee7653eb77a0b39852c2a676cf1a064e1676b8 Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Wed, 15 May 2024 14:11:16 +0200
Subject: [PATCH] Set FOLLY_HAVE_SO_TIMESTAMPING to 0

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 folly/net/NetOps.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/folly/net/NetOps.h b/folly/net/NetOps.h
index 23f23c145..7790848e9 100644
--- a/folly/net/NetOps.h
+++ b/folly/net/NetOps.h
@@ -120,7 +120,8 @@ struct mmsghdr {
 #define FOLLY_HAVE_MSG_ERRQUEUE 1
 #ifdef SCM_TIMESTAMPING
 #ifndef FOLLY_HAVE_SO_TIMESTAMPING
-#define FOLLY_HAVE_SO_TIMESTAMPING 1
+// See: https://github.com/conda-forge/folly-feedstock/blob/1f44286ea5c104cfead0d5cdbe85b52bde2b3066/recipe/build.sh#L10-L11C30
+#define FOLLY_HAVE_SO_TIMESTAMPING 0
 #endif
 #ifndef TCP_ZEROCOPY_RECEIVE
 #define TCP_ZEROCOPY_RECEIVE 35
-- 
2.44.0

