From 4f1f3f4646c6f68040ea78a8be4b7887076d857c Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Wed, 15 May 2024 14:47:36 +0200
Subject: [PATCH] Do not use PRId64

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 folly/Benchmark.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/folly/Benchmark.cpp b/folly/Benchmark.cpp
index ffe4b41e2..6bbb40312 100644
--- a/folly/Benchmark.cpp
+++ b/folly/Benchmark.cpp
@@ -500,7 +500,7 @@ class BenchmarkResultsPrinter {
               break;
             case UserMetric::Type::CUSTOM:
             default:
-              printf("  %*" PRId64, int(name.length()), ptr->value);
+              printf("  %*lld", (long long) int(name.length()), ptr->value);
           }
         } else {
           printf("  %*s", int(name.length()), "NaN");
-- 
2.44.0

