From c1ea3a2b60e2a153663abe5c3aa1c9bf51754b25 Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Fri, 17 May 2024 17:17:53 +0200
Subject: [PATCH] Deactivate libunwind features

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 folly/experimental/symbolizer/StackTrace.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/folly/experimental/symbolizer/StackTrace.cpp b/folly/experimental/symbolizer/StackTrace.cpp
index bf79dced7..bdc8edc50 100644
--- a/folly/experimental/symbolizer/StackTrace.cpp
+++ b/folly/experimental/symbolizer/StackTrace.cpp
@@ -24,6 +24,7 @@
 #include <folly/portability/Config.h>
 #include <folly/tracing/AsyncStack.h>
 
+#define FOLLY_HAVE_LIBUNWIND 0
 #if FOLLY_HAVE_LIBUNWIND
 // Must be first to ensure that UNW_LOCAL_ONLY is defined
 #define UNW_LOCAL_ONLY 1
-- 
2.45.1

