diff --git a/folly/debugging/symbolizer/StackTrace.cpp b/folly/debugging/symbolizer/StackTrace.cpp
index 31e6ceb..c4a7038 100644
--- a/folly/debugging/symbolizer/StackTrace.cpp
+++ b/folly/debugging/symbolizer/StackTrace.cpp
@@ -23,6 +23,7 @@
 #include <folly/Portability.h>
 #include <folly/portability/Config.h>
 
+#define FOLLY_HAVE_LIBUNWIND 0
 #if FOLLY_HAVE_LIBUNWIND
 // Must be first to ensure that UNW_LOCAL_ONLY is defined
 #define UNW_LOCAL_ONLY 1
