diff --git a/CMake/FollyCompilerMSVC.cmake b/CMake/FollyCompilerMSVC.cmake
index ad03039aa..42fd6fc22 100644
--- a/CMake/FollyCompilerMSVC.cmake
+++ b/CMake/FollyCompilerMSVC.cmake
@@ -118,10 +118,10 @@ function(apply_folly_compile_options_to_target THETARGET)
       /Zc:threadSafeInit # Enable thread-safe function-local statics initialization.
       /Zc:throwingNew # Assume operator new throws on failure.
 
+    PRIVATE
       /permissive- # Be mean, don't allow bad non-standard stuff (C++/CLI, __declspec, etc. are all left intact).
       /std:${MSVC_LANGUAGE_VERSION} # Build in the requested version of C++
 
-    PRIVATE
       /bigobj # Support objects with > 65k sections. Needed due to templates.
       /favor:${MSVC_FAVORED_ARCHITECTURE} # Architecture to prefer when generating code.
       /Zc:inline # Have the compiler eliminate unreferenced COMDAT functions and data before emitting the object file.
