diff --git i/CMake/FollyCompilerMSVC.cmake w/CMake/FollyCompilerMSVC.cmake
index ad03039aa..349fa2bbf 100644
--- i/CMake/FollyCompilerMSVC.cmake
+++ w/CMake/FollyCompilerMSVC.cmake
@@ -213,6 +213,7 @@ function(apply_folly_compile_options_to_target THETARGET)
       /wd4100 # Unreferenced formal parameter.
       /wd4459 # Declaration of parameter hides global declaration.
       /wd4505 # Unreferenced local function has been removed.
+      /wd4700 # Uninitialized local variable used.
       /wd4701 # Potentially uninitialized local variable used.
       /wd4702 # Unreachable code.
 
